docs/design/deviceshifu/design-deviceShifu-rtsp.md

```diff
@@ -0,0 +1,93 @@
+# deviceShifu Rtsp overall design
+
+deviceShifu-RTSP allows shifu connect to rtsp server and save the vedio into file. User can watch rtsp vedio using media player like VLC and get history vedio file from PV.
+
+## Goal
+
+Create a deviceShifu-RTSP using thired-party library and allow user to connect RTSP server, and record the rtsp stream into file like mp4 / flv and so on, user can export file out of k8s.
+
+## General Design
+
+deviceShifu-RTSP will receive RESTful style request like other, then transfer the request into RTSP method calls and send it to RTSP Server, meanwhile deviceShifu-RTSP provide a RTSP address which forward the RTSP stream from the target RTSP server.
+
+## Detailed Design
+
+### Protocol Specification
+
+deviceShifu-RTSP provide a RTSP server to forward RTSP stream and HTTP Server to handle Request.
+
+```go
+type ProtocolSettings struct {
+    MQTTSetting *MQTTSetting `json:"MQTTSetting,omitempty"`
+    ....
++   RTSPSetting *RTSPSetting `json:"RTSPSetting,omitempty"`
+}
+
+type RTSPSetting struct {
+    Username        string `json:"username,omitempty"`
+    PasswordSecret  string `json:"passwordSecret,omitempty"`
+    Record          bool   `json:"record,omitempty"`
+}
+```
+
+If user turn on Record, deviceShifu will Save the vedio into PV.(using ffmpeg or the others) 
```

Comment:
can we control the recording process to start/stop?
Also how is the file stored, i.e. split into small files or 1 file for the recording

Reply:
I wonder if we can use `TelemetryService` for recording to separate the encoding from deviceShifu

Reply:
telemetry is ok, If that I think we can do the record in telemetry and deviceShifu will have another container or another service in the cluster to handle the stream to record it. the start and stop is unnessary for securty and user to use

Reply:
I see, but the recording needs to be handled gracefully otherwise the only way to turn if off is to delete the pod. this will result in file corruption